packages/hint/src/lib/analyzer.ts

```diff
@@ -218,22 +219,23 @@ export class Analyzer {
                 throw new AnalyzerError(`Property 'content' is only supported in formatter local. Webhint will analyze the url ${url.href}`, AnalyzerErrorStatus.AnalyzeError);
             }
 
-            const engine = new Engine(this.configuration, this._resources);
+            this.engine = new Engine(this.configuration, this._resources);
 
-            this.configureEngine(engine, url.href, options);
+            this.configureEngine(this.engine, url.href, options);
 
             let problems: Problem[] | null = null;
 
             try {
                 if (options.targetStartCallback) {
                     await options.targetStartCallback({ url: url.href });
                 }
-                problems = await engine.executeOn(url, { content: target.content });
+                problems = await this.engine.executeOn(url, { content: target.content });
             } catch (e) {
                 throw new AnalyzerError(e, AnalyzerErrorStatus.AnalyzeError);
             } finally {
-                // TODO: Try if this is executed.
-                await engine.close();
+                if (this.engine) {
```

Comment:
I don't think this `if` is needed since `this.engine` is being directly assigned earlier in this function.

Reply:
done!